src/schnorrsig.rs

```diff
@@ -20,6 +20,12 @@ pub struct Signature([u8; constants::SCHNORRSIG_SIGNATURE_SIZE]);
 impl_array_newtype!(Signature, u8, constants::SCHNORRSIG_SIGNATURE_SIZE);
 impl_pretty_debug!(Signature);
 
+impl std::hash::Hash for Signature {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.0.hash(state)
+    }
+}
+
```

Comment:
Why not derive it?

Reply:
I recently learned that it is considered harmful to derive `Hash`/`PartialEq` and hand-implement the other and clippy complains about that (not sure if it would here, the `PartialEq` impl is behind a macro). The other question would be: why not derive most of the stuff currently in `impl_array_newtype!`?

Reply:
Because the stuff in impl_array_newtype needs to apply to array sizes for which `derive` does not work.

Why impl `Hash` here but not in `impl_array_newtype`?

Reply:
> Why impl `Hash` here but not in `impl_array_newtype`?

There already were some custom impls, updated the PR to remove them and instead implement `Hash` in the macro.